packages/frontend-api/src/Model/Mutation/Newsletter/NewsletterMutation.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Mutation\Newsletter;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Overblog\GraphQLBundle\Definition\Resolver\AliasedInterface;
+use Overblog\GraphQLBundle\Definition\Resolver\MutationInterface;
+use Overblog\GraphQLBundle\Validator\InputValidator;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Newsletter\NewsletterFacade;
+
+class NewsletterMutation implements MutationInterface, AliasedInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Newsletter\NewsletterFacade
+     */
+    protected NewsletterFacade $newsletterFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected Domain $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Newsletter\NewsletterFacade $newsletterFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(NewsletterFacade $newsletterFacade, Domain $domain)
+    {
+        $this->newsletterFacade = $newsletterFacade;
+        $this->domain = $domain;
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @param \Overblog\GraphQLBundle\Validator\InputValidator $validator
+     * @return bool[]
+     */
+    public function newsletterSubscribe(Argument $argument, InputValidator $validator): array
+    {
+        $validator->validate();
+
+        $input = $argument['input'];
+        $this->newsletterFacade->addSubscribedEmail($input['email'], $this->domain->getId());
```

Comment:
what about trimming data? it's not necessary?

Reply:
Yep, you're right. Fixed.